CodingTracker.MansoorAZafar/CodingTrackerLibrary/Controllers/CodingHabitManager.cs

```diff
@@ -0,0 +1,217 @@
+using CodingTrackerLibrary.Controllers.Database;
+using CodingTrackerLibrary.Models;
+using CodingTrackerLibrary.Views;
+
+namespace CodingTrackerLibrary.Controllers;
+
+internal class CodingHabitManager
+{
+    private DatabaseManager databaseManager = new();
+    private Reports? reports;
+    private bool? codingGoalExist;
+    private DateTime startingCodingGoalDate;
+    private int codingGoalHours;
+
+    public void DoMenuAction(MenuSelections chosenSelection)
+    {
+        Console.Clear();
+        int id = -1;
+
+        switch(chosenSelection)
+        {
+            case MenuSelections.update:
+                DataViewer.DisplayHeader("Updating a Coding Habit");
+                
+                this.UpdateCodingHabit(ref id);
```

Comment:
Remember that you can do 
`id = UpdateCodingHabit();`

and then change the signature of the method to:
`private int UpdateCodingHabit()`

Which is easier to read than:
`this.UpdateCodingHabit(ref id);`